source/extensions/original_ip_detection/custom_header/custom_header.cc

```diff
@@ -0,0 +1,46 @@
+#include "extensions/original_ip_detection/custom_header/custom_header.h"
+
+#include "common/network/utility.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace OriginalIPDetection {
+namespace CustomHeader {
+
+CustomHeaderIPDetection::CustomHeaderIPDetection(
+    const envoy::extensions::original_ip_detection::custom_header::v3::CustomHeaderConfig& config)
+    : header_name_(config.header_name()),
+      allow_trusted_address_checks_(config.common_config().allow_trusted_address_checks()) {
+
+  if (config.has_common_config() && config.common_config().reject_request_if_detection_fails()) {
+    Http::Code reject_code = Http::Code::Forbidden;
+
+    if (config.common_config().has_status_on_error()) {
+      reject_code = static_cast<Http::Code>(config.common_config().status_on_error().code());
+    }
+
+    reject_options_ = {reject_code, config.common_config().body_on_error(),
+                       config.common_config().details_on_error()};
+  }
+}
+
+Http::OriginalIPDetectionResult
+CustomHeaderIPDetection::detect(Http::OriginalIPDetectionParams& params) {
+  auto hdr = params.request_headers.get(Http::LowerCaseString(header_name_));
+  if (hdr.empty()) {
+    return {nullptr, false, reject_options_};
+  }
+  auto header_value = hdr[0]->value().getStringView();
+  try {
+    return {Network::Utility::parseInternetAddress(std::string(header_value)),
+            allow_trusted_address_checks_, absl::nullopt};
+  } catch (const EnvoyException&) {
+  }
```

Comment:
Since we are trying to remove exceptions from the data path, it would be good to not add new ones. Can we use/add a non-exception variant of this function?

Reply:
yes, there's `parseInternetAddressAndPortNoThrow`

Reply:
... err that's with port, I'll add one.